tests/test_circuit_graph.py

```diff
@@ -160,192 +120,92 @@ def test_dependence(self, queue, obs):
-        result = circuit.get_nodes(o_idxs)
-        assert result == [mock_graph.nodes[i] for i in o_idxs]
-
-    def test_get_ops(self, circuit, monkeypatch):
-        """Test that `get_ops` fetches the correct ops based on given nodes."""
-        mock_nodes = MagicMock()
-        mock_ops = MagicMock()
-        mock_get_nodes = MagicMock()
-        mock_get_nodes.return_value = [mock_nodes.a, mock_nodes.b]
-        monkeypatch.setattr(circuit, "get_nodes", mock_get_nodes)
-
-        result = circuit.get_ops(mock_ops)
-        mock_get_nodes.assert_called_once_with(mock_ops)
-        assert result == [mock_nodes.a["op"], mock_nodes.b["op"]]
-
-    def test_update_node(self, circuit, monkeypatch):
-        """Test that `nx.set_node_attributes` is called correctly when updating a node using
-        `update_node` and a given node and op."""
-        mock_command = MagicMock()
-        mock_nx = MagicMock()
-
-        monkeypatch.setattr("pennylane.circuit_graph.nx", mock_nx)
-        monkeypatch.setattr("pennylane.circuit_graph.Command", mock_command)
-
-        node = MagicMock()
-        op = MagicMock()
-
-        circuit.update_node(node, op)
-
-        mock_nx.set_node_attributes.assert_called_once_with(
-            circuit._graph,
-            {
-                node["idx"]: {
-                    **mock_command(
-                        name=op.name, op=op, return_type=op.return_type, idx=node["idx"]
-                    )._asdict()
-                }
-            },
-        )
+    def test_update_node(self, ops):
+        """Changing nodes in the graph."""
+
+        circuit = CircuitGraph(ops, {})
+        new = qml.RX(0.1, wires=0, do_queue=False)
+        circuit.update_node(ops[0], new)
+        assert circuit.operations[0] is new
 
     def test_observables(self, circuit, obs):
         """Test that the `observables` property returns the list of observables in the circuit."""
         assert circuit.observables == obs
 
-    def test_observable_nodes_and_operation_nodes(self, circuit, monkeypatch):
-        """Test that the `observable_nodes` and `operation_nodes` methods return the correct
-        values."""
-        mock_graph = MagicMock()
-        mock_graph.nodes.values.return_value = [
-            {"return_type": None, "idx": 0},
-            {"return_type": MagicMock(), "idx": 1},
-        ]
-        monkeypatch.setattr(circuit, "_graph", mock_graph)
-        assert circuit.observable_nodes == [mock_graph.nodes.values()[1]]
-        assert circuit.operation_nodes == [mock_graph.nodes.values()[0]]
-
     def test_operations(self, circuit, queue):
         """Test that the `operations` property returns the list of operations in the circuit."""
         assert circuit.operations == queue
 
-    def test_get_names(self, circuit):
-        result = circuit.get_names(range(9))
-        expected = ["RX", "RY", "RZ", "CNOT", "Hadamard", "CNOT", "PauliX", "PauliX", "Hermitian"]
-        assert result == expected
-
```

Comment:
I suppose this test is redundant because the `get_names` method is redundant? I.e., you can now just do `[circuit.operations[i].name for i in range(9)]`?

Reply:
Right.